src/data/scripts/campaign/MPC_coronaResistScript.kt

```diff
@@ -55,24 +99,32 @@ open class MPC_coronaResistScript(
         }
     }
 
+    private fun affectFleetMember(member: FleetMemberAPI){
+        member.stats.dynamic.getStat(Stats.CORONA_EFFECT_MULT).modifyMult(
+            "${entity.id}_MPCspecialCoronaResistance",
+            coronaResistance,
+            "Unknown corona resistance"
+        )
+    }
+
+    private fun unAffectFleetMember(member: FleetMemberAPI){
+        member.stats.dynamic.getStat(Stats.CORONA_EFFECT_MULT).unmodify("${entity.id}_MPCspecialCoronaResistance")
+    }
+
     private fun affectFleet(fleet: CampaignFleetAPI, days: Float) {
         fleet.counterTerrainMovement(days, terrainMovementDivisor) // this doesnt seem to work very well either, its inconsistant between fleets
```

Comment:
this right here is the tempermental piece of code, it does its best to approximate the current movement generated by whatever is moving the fleet

this is kinda why i opted not to add a interval because i just got it to work ™️ and didnt want to anger the code gods by trying an optimizationn or two

Reply:
affectFleet is called by iterateThroughFleets,, which is not inside the interval-portion. the interval is only for checking "should this ship/fleet still have the corona modifier?". I think I encountered an issue like what you are describing, but I havent encountered it again since I split those portions off.

Reply:
ah, ok, its probably fine then